packages/gateway/src/features/search/root-resolvers.ts

```diff
@@ -190,6 +205,55 @@ export const resolvers: GQLResolver = {
             searchResult.body.hits.hits) ||
           []
       }
+    },
+    async searchRecord(_, searchCriteria: IAdvancedSearchParam, authHeader) {
+      if (authHeader && !hasScope(authHeader, 'recordsearch')) {
+        return await Promise.reject(new Error('User does not have permission'))
+      }
+
+      const payload = getTokenPayload(authHeader.Authorization)
+      const system = await getSystem({ systemId: payload.sub }, authHeader)
+
+      const getTotalRequest = await getMetrics(
+        '/advancedSearch',
+        {},
+        authHeader
+      )
+
+      if (getTotalRequest.total >= system.settings.dailyQuota) {
+        return await Promise.reject(new Error('Daily search quota exceeded'))
+      }
+
+      await postMetrics('/advancedSearch', {}, authHeader)
+
+      const searchResult: ApiResponse<ISearchResponse<any>> =
```

Comment:
Before posting to metrics, we should probably get the results from search. Because if the search fails a few times let's say due to invalid request params, then the user won't be able to continue with the search because of the quota.